test/verify/machineslib.py

```diff
@@ -686,16 +687,26 @@ def verify_disk_added(self):
                 b.wait_in_text("#vm-{0}-disks-{1}-device".format(self.vm_name, self.expected_target), "disk")
 
                 # Detect volume format
-                detect_format_cmd = "virsh dumpxml {0} | xmllint --xpath '/domain/devices/disk[@type=\"{1}\"]/driver[@type=\"qcow2\"]' -"
+                detect_format_cmd = "virsh vol-dumpxml {0} {1} | xmllint --xpath '/volume/target/format' -".format(self.volume_name, self.pool_name)
```

Comment:
Is is correct to check the format of the volume instead of disk?
This would make testing format type pointless for "Use existing" option in attaching disk dialog, as there we only create attach a disk into format domain XML, a with this change it would be left untested.

Reply:
It's correct if we want to test that the format the user inserts from the UI get's reflected in the volume XML.
The disk driver type won't always have the value of the format chosen from the dialog; See disk type pools, where volume format would be some partition table type, though disk driver type is raw.

Reply:
Oh I see. There is like 5 pool types which will always have 'volume format type' == 'disk format type', so we could add some test case where we test disk format only for those pool types. But that's just an idea for future PR, not this PR.